In [1]:
library(fBasics)

Warning message:
"package 'fBasics' was built under R version 4.2.2"


##### a)  Hãy ước lượng hàm sản xuất có dạng Cobb- Douglas. Phương sai của sai số có đồng đều không?

In [4]:
df = read.table("ch3bt4.txt",header = TRUE)
hq <- lm(data = df,log(Y)~log(K)+log(L))
summary(hq)


Call:
lm(formula = log(Y) ~ log(K) + log(L), data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.56455 -0.21716  0.03205  0.16651  0.42318 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  9.77025    0.22857  42.745  < 2e-16 ***
log(K)       0.52370    0.09376   5.586 3.28e-05 ***
log(L)       0.69301    0.14054   4.931 0.000127 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.282 on 17 degrees of freedom
Multiple R-squared:  0.7814,	Adjusted R-squared:  0.7557 
F-statistic: 30.39 on 2 and 17 DF,  p-value: 2.436e-06


Kiểm định phương sai có đồng đều không (BP Test)
$\begin{cases} 
\text{H0: phương sai là không đổi}\\
\text{H1: phương sai là thay đổi.}
\end{cases}$

In [7]:
library(lmtest)
bptest(hq)


	studentized Breusch-Pagan test

data:  hq
BP = 0.35168, df = 2, p-value = 0.8388


Do giá trị p-value = 0.8388 < 5% nên chúng ta có bằng chứng thống kê bắc bỏ Ho

##### c) Hiệu quả sản xuất không đổi theo quy mô? 

In [9]:
library(car)
linearHypothesis(hq,c("log(K)+log(L)=1"))

Res.Df RSS      Df Sum of Sq F       Pr(>F)   
1 18     1.494548 NA        NA      NA        NA
2 17     1.352226  1 0.1423218 1.78925 0.1986389

##### d) Khoảng tin cậy

In [23]:
confint(hq,level = 0.9)

5 %       95 %      
(Intercept) 9.3726319 10.1678699
log(K)      0.3606023  0.6867965
log(L)      0.4485214  0.9374895

$\hat\beta_1+ \hat\beta_2-t_{\alpha/2}se(\hat\beta_1+ \hat\beta_2)<\beta_1+ \beta_2<\hat\beta_1+ \hat\beta_2+t_{\alpha/2}se(\hat\beta_1+ \hat\beta_2)$

In [25]:
vcov(hq)
var2 <- vcov(hq)[2,2]
var3 <- vcov(hq)[3,3]
cov23 <- vcov(hq)[3,2]

(Intercept) log(K)       log(L)      
(Intercept)  0.05224348 -0.012182545 -0.023211362
log(K)      -0.01218254  0.008790028 -0.001147648
log(L)      -0.02321136 -0.001147648  0.019751448

In [26]:
se <- sqrt(var2+var3+2*cov23)
beta2 <- 0.52370 
beta3 <- 0.69301

In [27]:
can_tren <- (beta2+beta3)+qt(0.95,length(df$K)-3)*se
can_duoi <- (beta2+beta3)-qt(0.95,length(df$L)-3)*se

In [28]:
can_tren

[1] 1.498538

In [29]:
can_duoi

[1] 0.934882

##### e) Dự báo với 2 quan sát sau mẫu: K=(50,45), L=(35,32).

$\beta_1 = \beta_2$

In [11]:
library(car)
linearHypothesis(hq,c("log(K)=log(L)"))

Res.Df RSS      Df Sum of Sq  F         Pr(>F)   
1 18     1.426166 NA         NA        NA        NA
2 17     1.352226  1 0.07393951 0.9295573 0.3484935

$\beta_1 = \beta_2\ và\ \beta_2=0.3$

In [12]:
linearHypothesis(hq,c("log(K)+log(L)=1","log(L)=0.3"))

Res.Df RSS      Df Sum of Sq F       Pr(>F)   
1 19     2.188990 NA        NA      NA        NA
2 17     1.352226  2 0.8367639 5.25984 0.0166666

##### f) Dự báo với 2 quan sát sau mẫu: K=(50,45), L=(35,32).

In [17]:
K <- c(50,45)
L <- c(35,32)
predic1 <- predict(hq,data.frame(K,L))
predic1

1        2 
14.28285 14.16557

In [18]:
predic2 <- predict(hq,data.frame(K,L),interval="confidence",level=0.95)
predic2

fit      lwr      upr     
1 14.28285 13.48183 15.08387
2 14.16557 13.39632 14.93482